discovery-provider/plugins/notifications/src/processNotifications/mappers/announcement.ts

```diff
@@ -13,84 +13,112 @@ type AnnouncementNotificationRow = Omit<NotificationRow, 'data'> & {
+        name: string
+        is_deactivated: boolean
+      }> = await this.dnDB
+        .select('user_id', 'name', 'is_deactivated')
+        .from<UserRow>('users')
+        // query by last id seen
+        .where('user_id', '>', offset)
+        .andWhere('is_current', true)
+        .andWhere('is_deactivated', false)
+        .orWhere((inner) =>
+          inner.where('user_id', '=', offset).andWhere('is_current', true)
+        )
+        .andWhere('is_deactivated', false)
+        // order by established index, this keeps perf constant
+        .orderBy(['user_id', 'is_current', 'txhash'])
+        .limit(page_count)
+
+      offset = res[res.length - 1].user_id + 1
+      const elapsed = new Date().getTime() - start
+      console.log(
+        `count: ${count} offset: ${offset} from: [${res[0].user_id}:${res[res.length - 1].user_id}] queried in ${elapsed} ms`
       )
-      // If the user has devices to the notification to, proceed
-      if (
-        userNotificationSettings.shouldSendPushNotification({
-          receiverUserId: userId
-        })
-      ) {
-        const devices: Device[] = userNotificationSettings.getDevices(userId)
-        // If the user's settings for the follow notification is set to true, proceed
 
-        await Promise.all(
-          devices.map((device) => {
-            return sendPushNotification(
-              {
-                type: device.type,
-                badgeCount: userNotificationSettings.getBadgeCount(userId) + 1,
-                targetARN: device.awsARN
-              },
-              {
-                title: this.notification.data.title,
-                body: this.notification.data.short_description,
-                data: {
-                  id: `timestamp:${this.getNotificationTimestamp()}:group_id:${
-                    this.notification.group_id
-                  }`,
-                  type: 'Announcement'
-                }
-              }
-            )
-          })
+      const validReceiverUserIds = res.map((user) => user.user_id)
+      for (const userId of validReceiverUserIds) {
+        const userNotificationSettings = await buildUserNotificationSettings(
+          this.identityDB,
+          [userId]
         )
-        await this.incrementBadgeCount(userId)
-      }
-      if (
-        userNotificationSettings.shouldSendEmail({ receiverUserId: userId })
-      ) {
-        // TODO: Send out email
-      }
+        // If the user has devices to the notification to, proceed
+        if (
+          userNotificationSettings.shouldSendPushNotification({
+            receiverUserId: userId
+          })
+        ) {
+          const devices: Device[] = userNotificationSettings.getDevices(userId)
+          // If the user's settings for the follow notification is set to true, proceed
+
+          await Promise.all(
+            devices.map((device) => {
+              // this may get rate limited by AWS
+              return sendPushNotification(
```

Comment:
i think given it's a bunch of users we're notifying it would be great if we could use `PublishBatchCommand` from aws sdk. if you look in sns.ts in the notif plugin, there's a function for it already, but you may have to add a `sendPushNotificationBatch` function that calls it separately. and then we can create an array of batch messages to send and only add to it if the user's settings says shouldSendPushNotification is true

hopefully this makes sense but pls lmk if not!

Reply:
sure that sounds good to me 👍

Reply:
hmm so batches seem to only work with topics which we're not using yet, lets use it in a future ticket? https://awscli.amazonaws.com/v2/documentation/api/2.4.18/reference/sns/publish-batch.html

Reply:
ah ok thanks for looking into that, ill make a ticket for it